pkg/receive/hashring.go

```diff
@@ -165,19 +165,19 @@ func newMultiHashring(cfg []HashringConfig) Hashring {
 // Which hashring to use for a tenant is determined
 // by the tenants field of the hashring configuration.
 // The updates chan is closed before exiting.
-func HashringFromConfig(ctx context.Context, updates chan<- Hashring, cw *ConfigWatcher) {
-	go cw.Run(ctx)
+func HashringFromConfig(ctx context.Context, updates chan<- Hashring, cw *ConfigWatcher) error {
 	defer close(updates)
+	go cw.Run(ctx)
 
 	for {
 		select {
 		case cfg, ok := <-cw.C():
 			if !ok {
-				return
+				return errors.New("hashring config watcher stopped")
```

Comment:
Can we make one tiny change here and add `unexpectedly` to the end? Just to highlight that this should only ever occur if the config watcher stopped irregularly. The only other exit scenario is a planned exit when the context is cancelled.